data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -12,28 +12,45 @@ import scala.collection.mutable
 import scala.math.Ordered.orderingToOrdered
 
 
-class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyEntityPayload, LandyEntityPayload] {
-  override val vertices: VertexRDD[LandyEntityPayload] = graph.vertices
-  override val edges: EdgeRDD[LandyEntityPayload] = graph.edges
-  override val triplets: RDD[EdgeTriplet[LandyEntityPayload, LandyEntityPayload]] = graph.triplets
+class Landy(graph: LandyAttributeGraph) extends TemporalGraph {
+  val vertices: VertexRDD[LandyEntityPayload] = graph.vertices
+  val edges: EdgeRDD[LandyEntityPayload] = graph.edges
 
-  override def snapshotAtTime(instant: Instant): Graph[LandyEntityPayload, LandyEntityPayload] = {
+  def snapshotAtTimeLandy(instant: Instant): Graph[LandyEntityPayload, LandyEntityPayload] = {
     val vertices = localVertices
       .filter(vertex =>
         vertex._2.validFrom < instant &&
           vertex._2.validTo >= instant
       )
       .map(vertex => (vertex._2.id, vertex._2))
 
-    val edges = this.edges.filter(edge =>
+    val edges = this.graph.edges.filter(edge =>
       edge.attr.validFrom < instant &&
         edge.attr.validTo >= instant
     )
     Graph(vertices, edges)
   }
 
+  override def snapshotAtTime(instant: Instant): Snapshot = {
+    val vertices = localVertices
+      .filter(vertex =>
+        vertex._2.validFrom < instant &&
+          vertex._2.validTo >= instant
+      )
```

Comment:
```suggestion
      .filter(vertex =>
        vertex._2.interval.contains(instant)
      )
```
Blir ikke heeelt det samme fordi contains er  inklusiv, dvs.       validfrom <= instant<= validto
Men vi får ev. endre contains-logikken